torch/csrc/distributed/c10d/CUDASymmetricMemoryOps.cu

```diff
@@ -1145,16 +1146,29 @@ at::Tensor stream_write_value32_(
   c10::cuda::CUDAGuard guard(input.device());
 
 #if !defined(USE_ROCM) && defined(PYTORCH_C10_DRIVER_API_SUPPORTED)
-  auto driver_api = c10::cuda::DriverAPI::get();
-  // According to the documentation of CUstreamWriteValue_flags,
-  // cuStreamWriteValue32 will provide a memory fence before the write, which
-  // has similar semantics to __threadfence_system() but is scoped to the
-  // stream rather than a CUDA thread.
-  C10_CUDA_DRIVER_CHECK(driver_api->cuStreamWriteValue32_(
+  // --- Begin dynamic loading of cuStreamWriteValue32 ---
+  typedef CUresult (CUDAAPI *PFN_cuStreamWriteValue32)(
+      CUstream hStream, CUdeviceptr addr, uint32_t value, unsigned int flags);
+  static PFN_cuStreamWriteValue32 pfn_cuStreamWriteValue32 = nullptr;
+  static std::once_flag load_cuStreamWriteValue32_flag;
```

Comment:
Please don't use `std::once_flag` but instead rely on C++11 built in mechanism by using something like
```
static auto pfn_cuStreamWriteValue32 = []() {
   // Move all the initialization logic here, which is guaranteed to be called once the first time function is called
}();